src/main/java/server/HttpServer.java

```diff
@@ -1,30 +1,45 @@
 package com.td.HttpServer;
 import java.io.IOException;
 import java.net.ServerSocket;
+
 public class HttpServer{
 
-  private static ArgumentParser arguments;
+  private ServerSocket serverSocket;
+  private HttpServerRunner httpServerRunner;
+  private CustomHandlerRouter customHandlerRouter;
+  private FileIO fileIO;
 
-  public static void main(String[] args) {
-    arguments = new ArgumentParser(args);
-    ServerSocket serverSocket;
+  public HttpServer(int portNumber, String directory) {
     try {
-      serverSocket = new ServerSocket(arguments.getPortNumber());
+      serverSocket = new ServerSocket(portNumber);
     }
     catch (IOException e) {
-      System.out.println("\nPort number " + arguments.getPortNumber() + " can not be opened. It may be in use by another application.\n");
+      System.out.println("\nPort number " + portNumber + " can not be opened. It may be in use by another application.\n");
       return;
     }
     HttpRequestParser httpRequestParser = new HttpRequestParser();
-    FileIO fileIO = new FileIO(arguments.getDirectory());
+    fileIO = new FileIO(directory);
     DefaultHandlerRouter defaultHandlerRouter = new DefaultHandlerRouter(fileIO);
-    CustomHandlerRouter customHandlerRouter = new CustomHandlerRouter(defaultHandlerRouter);
+    customHandlerRouter = new CustomHandlerRouter(defaultHandlerRouter);
     HttpRequestBuilder httpRequestBuilder = new HttpRequestBuilder(httpRequestParser);
     HttpResponseWriter httpResponseWriter = new HttpResponseWriter();
     ConnectionProcessRunnerMultiThread connectionProcessRunner = new ConnectionProcessRunnerMultiThread();
-    HttpServerRunner httpServerRunner = new HttpServerRunner(serverSocket, connectionProcessRunner, customHandlerRouter, httpRequestBuilder, httpResponseWriter);
+    httpServerRunner = new HttpServerRunner(serverSocket, connectionProcessRunner, customHandlerRouter, httpRequestBuilder, httpResponseWriter);
+  }
+
+  public void run() {
+    httpServerRunner.run();
```

Comment:
Is this method blocking? Will the following two statements get printed out only when the server stops (and returns from `run`), put differently never?